plugins/buildingplan.cpp

```diff
@@ -911,26 +911,75 @@ IMPLEMENT_VMETHOD_INTERPOSE(buildingplan_query_hook, render);
 IMPLEMENT_VMETHOD_INTERPOSE(buildingplan_place_hook, render);
 IMPLEMENT_VMETHOD_INTERPOSE(buildingplan_room_hook, render);
 
+DFHACK_PLUGIN_IS_ENABLED(is_enabled);
+
+static void setSetting(std::string name, bool value);
+
+static bool isTrue(std::string val)
+{
+    return val == "on" || val == "true" || val == "y" || val == "yes";
+}
+
 static command_result buildingplan_cmd(color_ostream &out, vector <string> & parameters)
 {
-    if (!parameters.empty())
+    if (parameters.empty())
+        return CR_OK;
+
+    std::string cmd = toLower(parameters[0]);
+
+    if (cmd.size() >= 1 && cmd[0] == 'v')
+    {
+        out << "buildingplan version: " << PLUGIN_VERSION << endl;
+    }
+    else if (parameters.size() >= 2 && cmd == "debug")
     {
-        if (parameters.size() == 1 && toLower(parameters[0])[0] == 'v')
+        show_debugging = isTrue(toLower(parameters[1]));
+        out << "buildingplan debugging: " <<
+                ((show_debugging) ? "enabled" : "disabled") << endl;
+    }
+    else if (cmd == "set")
+    {
+        if (!is_enabled)
         {
-            out << "Building Plan" << endl << "Version: " << PLUGIN_VERSION << endl;
+            out << "ERROR: buildingplan must be enabled before you can read or "
```

Comment:
Errors could use `out.printerr()` (which doesn't support `<<` syntax, but it looks like both messages are just raw strings)

Reply:
Changed it all. And for the better. Have I ever mentioned how much I dislike c++ stream insertion for string output?

Reply:
They don't translate quite as directly as I'd like - specifically, `print()`/`printerr()` require newlines in the messages (I took care of adding the ones that were missing).